### DISCUSSION TITLE
Windows version

### DISCUSSION THREAD

**Discussion by eskimos20** [cagnulein/qdomyos-zwift#2904] [Q&A]
Hi! 

I really like this program but have some things to discuss. 

I´ve so far tried the app on:

1. Android Tablet and this is working best for me (only that connections take way longer than normal, about 1-2 minutes compared to regular setup when my Taxc Neo2 connects to my zwift app in about 10 seconds)
2. Linux Works but the process of installing from source takes very long time :) 
3. Windows 11 and the exe is not working. At least not on Windows 11. It just hangs all the time and I´ve also read in the other thread that people have the same problems that I do. And if it does not hang then you could connect to the bluetooth devices, it looks OK but nothing happens when you start to pedal.  So there must be some work to do here still :) I have the vc 64 bit installed aswell.

I´m a devloper myself so trying it on different plattforms have been great. 

I have question. My Zwift application is on my Windows 11 machine with a Bluetooth dongle. This setup works very well when I connection my heart rate monitor and my Taxc directly to it. But my question is this:

If the Windows QZ App is installed on the same machine as the Zwift game, could I connect my heart rate monitor, my trainer and my zwift click to the running QZ App on Windows 11 and then the App would communicate with the local Zwift installation ? Like the picture below. 
![QZ Fitness Windows](https://github.com/user-attachments/assets/e2711afb-791f-4445-b374-7a45e8440bf6)

/Eskimos20

**Comment by cagnulein** [comment]
> 1. Android Tablet and this is working best for me (only that connections take way longer than normal, about 1-2 minutes compared to regular setup when my Taxc Neo2 connects to my zwift app in about 10 seconds)

qz takes 2 minutes to connect to the trainer? show me a debug log and i can check it https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F



> 2\. Linux Works but the process of installing from source takes very long time :)

i'm planning to release binary too



> 3\. Windows 11 and the exe is not working. At least not on Windows 11. It just hangs all the time and I´ve also read in the other thread that people have the same problems that I do. And if it does not hang then you could connect to the bluetooth devices, it looks OK but nothing happens when you start to pedal.  So there must be some work to do here still :) I have the vc 64 bit installed aswell.

yes windows 11 broke something. Did you try to run qz with the compatibility mode to windows xp or 7?



> If the Windows QZ App is installed on the same machine as the Zwift game, could I connect my heart rate monitor, my trainer and my zwift click to the running QZ App on Windows 11 and then the App would communicate with the local Zwift installation ? Like the picture below.

yes using the wahoo dircon device. but as i said you have first to fix the windows 11 issue or with the compatibility mode, if it works, or with a virtual box

**Reply by eskimos20** [reply]
I tried Windows app now in compability mode for Windows 7. Now It works a bit better. But you dont dare to change to many things. Like going into settings might crash the app. I´ll check the debug log for Android a bit later. 

Thanks for the quick answer :)

Wahoo dircon, I think that is for Wahoo bikes like kickr.. I have a Taxc Neo2. 

/Eskimos20

**Reply by cagnulein** [reply]
> Wahoo dircon, I think that is for Wahoo bikes like kickr.. I have a Taxc Neo2.

nope it's the way qz shows itself on zwift :)



> I tried Windows app now in compability mode for Windows 7. Now It works a bit better. But you dont dare to change to many things. Like going into settings might crash the app.

strange, do you have a debug log as well?

**Reply by eskimos20** [reply]
Ah I understand so then It should work on the same device and that sounds great. Just need to find the bug on windows then.

I would like to update you on the connection issue, maybe I was a bit unclear.

1. My QZ App on android tablet takes about 10 seconds to connect to the Trainer and the Heart Rate Monitor
2. The connection between the Android tablet --> Zwift running on a Win 11 machine bith bluetooth takes about 1-2 minutes, normally this step takes about 10 seconds if the QZ app is not used. 

I ran it now with debugging enabled. However tha app is so unstable that its hard to get to settings/experimental/debug choice :D

But managed to do it! 

Hope you find something. 
[Debug.txt](https://github.com/user-attachments/files/18126572/Debug.txt)


/Eskimso20

**Reply by cagnulein** [reply]
> 1. My QZ App on android tablet takes about 10 seconds to connect to the Trainer and the Heart Rate Monitor

be sure both trainer and heart rate are on advertisting. the 10 seconds is the timeout when a device is not connecting



> 2\. The connection between the Android tablet --> Zwift running on a Win 11 machine bith bluetooth takes about 1-2 minutes, normally this step takes about 10 seconds if the QZ app is not used.

qz is bringing up the bridge immediately. which bluetooth dongle are you using ?

**Comment by eskimos20** [comment]
Asus BT500, works great all the time.

Den fre 13 dec. 2024 14:05Roberto Viola ***@***.***> skrev:

**Reply by cagnulein** [reply]
BT400 is the best IMHO.
Anyway on PC you should use the Wahoo dircon protocol, in order to have best performance

**Comment by eskimos20** [comment]
How do i use the dircon protocol, is it an dongle?

Den fre 13 dec. 2024 14:10Roberto Viola ***@***.***> skrev:

**Reply by cagnulein** [reply]
no it's the wahoo kickr device that you see on zwift!

**Comment by eskimos20** [comment]
[debug-Mon_Jan_6_10_08_27_2025.log](https://github.com/user-attachments/files/18317332/debug-Mon_Jan_6_10_08_27_2025.log)
Hi! 

Do you have a working version for windows 11 now ? 

I tried the latest nightly but it is just:

Mon Jan 6 10:09:52 2025 1736154592555 Debug: devices/bluetooth.cpp void bluetooth::debug(const QString&) "BTLE scanning finished"
"BTLE scanning finished"

It cant find any bluetooth devices, tried 2 sticks, BT500 and BT400. 

The app seems better, not hanging in win 11 but just cant find bluetooth :)

**Comment by cagnulein** [comment]
You have to pair the Bluetooth device on the windows settings for this

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 gen 2025 alle 10:16 eskimos20 ***@***.***> ha
scritto:

**Comment by eskimos20** [comment]
Ok got it. It kind of worked. Watts cadence showed up in QZ app but could not get it to send that to zwift. But I will stick with docker and linux for now. Thanks anyway :)

**Comment by cagnulein** [comment]
You have to use the wahoo kickr interface on zwift

Yes Linux is better  :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 gen 2025 alle 10:57 eskimos20 ***@***.***> ha
scritto:

**Comment by eskimos20** [comment]
I used the wahoo kickr interface. It was the only available. Could not see
any other 🫤

Den mån 6 jan. 2025 11:04Roberto Viola ***@***.***> skrev:

**Reply by cagnulein** [reply]
Yes that one is perfect. What issue did you have precisely?

**Comment by eskimos20** [comment]
The issue is that nothing happens when i pedal in zwift. The qz app
register the watt and cadence but not zwift even when connected.

Den mån 6 jan. 2025 13:46Roberto Viola ***@***.***> skrev:

**Reply by cagnulein** [reply]
That's another issue: did you try to delete the know device file in the zwift folder? Probably it was a cache issue on zwift

**Comment by eskimos20** [comment]
No i did not do that. I will try that. Does the windows version work with
-no-gui -no-log -zwft_play_emulator option?

Den mån 6 jan. 2025 13:53Roberto Viola ***@***.***> skrev:

**Comment by cagnulein** [comment]
I didn't try to run in this way but it should work too

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 gen 2025 alle 13:57 eskimos20 ***@***.***> ha
scritto:

**Comment by eskimos20** [comment]
Ok I now tried with: .\qdomyos-zwift.exe -no-gui -zwift_play_emulator -no-log -no-console

What happens is that the trainer gets connected and the gui does not show like expected. But it writes a log, it writes to console and I cant connect my zwift click in Zwift. 

I can see that zwift is connected via Wahoo interface but when pedeling nothing happens. If I shutdown QZ terminal the connection is lost in Zwift so it seems to be connected to the right source. 

I will send you the log.

I am running Zwift and QZ app on the same computer. Win 11 Professional. I am running QZ app as Admin via terminal.

[debug-Mon_Jan_6_14_04_00_2025.log](https://github.com/user-attachments/files/18319843/debug-Mon_Jan_6_14_04_00_2025.log)

**Reply by cagnulein** [reply]
click on zwift doesn't connect? so that's a zwift issue :) i mean it's qz not involved :)
i got the issue about the metrics, i will check this in the next days, thanks

**Reply by eskimos20** [reply]
I mean when running Linux it the options for me at least has to be:

Power device first connected to QZPI.
Cadence connected to QZPI
Controller conntetod to QZPI 

only then will it connect to Zwift Click... 

after that I will switch to Wahoo Kickr for Power and Cadence.. the rest stays at QZPI.

But when running Windows QZPI is not accessible.

**Comment by cagnulein** [comment]
got it anyway there is a biggest problem first: the metrics are not coming
to qz, at least not in the log that you sent me. so first of all that one
must be addressed.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 6 gen 2025 alle ore 14:24 eskimos20 ***@***.***>
ha scritto:

**Comment by Zwifter458** [comment]
Which version is best for windows 10?

**Comment by cagnulein** [comment]
Msvc no python

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 12 feb 2025 alle 19:14 Zwifter458 ***@***.***>
ha scritto:

**Comment by Zwifter458** [comment]
I just checked through the options (there are quite a lot of them) and it threw me out that the trial period is over. How long is the trial period? I just want to try virtual shifting and how it works, before i go into any patreon stuff

**Comment by cagnulein** [comment]
It's 15 minutes and you can restart as many as you want

**Comment by cagnulein** [comment]
Describe me also your setup and what you would like to obtain

**Comment by Zwifter458** [comment]
I have a windows PC with Tacx neo 2T smart trainer + zwift play and all i wanna do is to try virtual shifting in zwift.
And..I have some considerations regarding use of your software, but id like to discus that more privately, but i cant find a way to contact you... are you on some discord server?

**Comment by cagnulein** [comment]
Sure https://discord.gg/ABB79mmk

For the windows build remeber to use the dircon branch for windows so you
will be able to use qz and zwift on the same device
